include/particle.h

```diff
@@ -105,17 +105,25 @@ class Particle : public ParticleBase<Tdim> {
   //! \param[in] porosity Porosity of particle
   bool assign_porosity(double porosity) override;
 
-  //! Return volume
+  //! Return volume of specified phase
   //! \param[in] phase Index corresponding to the phase
   double volume(unsigned phase) const override { return phase_volume_(phase); }
 
+  //! Return total volume
+  double total_volume() const override { return volume_; }
```

Comment:
Let's use function overloading here.
We will have two functions with same name as
`double volume(unsigned  phase) const override {return volume_*volume_fraction_(phase); }`
`double volume() const override { return volume_; }`